### PR TITLE
Added requirement for ICU Data

### PR DESCRIPTION
**Pull request by SilverFire**
As a result of segfaults #10152, #10171

### CODE REVIEW

File: framework/requirements/requirements.php [hunk 1]
```diff
@@ -73,6 +73,17 @@
         INTL_ICU_VERSION . '.'
     ),
     array(
+        'name' => 'ICU Data version',
+        'mandatory' => false,
```
Comment (SilverFire) [inline framework/requirements/requirements.php]: I think we can switch `mandatory` to `true` in both ICU and ICU Data requirements

File: framework/UPGRADE.md [hunk 2]
```diff
@@ -17,6 +17,17 @@ php composer.phar global require "fxp/composer-asset-plugin:~1.1.0"
 Upgrade from Yii 2.0.6
 ----------------------
 
+* Added new requirement: ICU Data version >= 49.1. Please, ensure that your environment has ICU data installed and
+up to date to prevent unexpected behavior or crashes.
+
+ > Tip: The easiet way to check: `php --info | grep ICU`
```
Comment (samdark) [inline framework/UPGRADE.md]: It's not:

1. We have requirements script.
2. The common mistake is to use different `php.ini` in CLI / fpm / apache module.

File: framework/requirements/requirements.php (hunk 1 above)
Comment (samdark) [inline framework/requirements/requirements.php]: I don't think so. If you're developing English-only website you're fine w/o intl at all.

File: framework/requirements/requirements.php [hunk 3]
```diff
@@ -73,6 +73,17 @@
...
+        'condition' => defined('INTL_ICU_DATA_VERSION') && version_compare(INTL_ICU_DATA_VERSION, '49.1', '>='),
+        'by' => '<a href="http://www.php.net/manual/en/book.intl.php">Internationalization</a> support',
+        'memo' => 'ICU Data 49.1 or higher is required when you want to use <code>#</code> placeholder in plural rules
+        (for example, plural in
+        <a href=\"http://www.yiiframework.com/doc-2.0/yii-i18n-formatter.html#asRelativeTime%28%29-detail\">
+        Formatter::asRelativeTime()</a>) in the <code>yii\i18n\Formatter</code> class. Your current ICU Data version is ' .
+        INTL_ICU_DATA_VERSION . '.'
```
Comment (cebe) [inline framework/requirements/requirements.php]: defined check needed for this too.

File: framework/requirements/requirements.php [hunk 4]
```diff
@@ -73,6 +73,17 @@
         INTL_ICU_VERSION . '.'
```
Comment (cebe) [inline framework/requirements/requirements.php]: same here btw.

**Comment by SilverFire** [comment]
https://github.com/yiisoft/yii2/commit/9d79038485ea4ca79a6137b3d09bb271bc97c9d4

**Comment by SilverFire** [comment]
Thank you!

**Comment by samdark** [comment]
:+1:

**Pull request closed without merging**